### PR TITLE
Cache Initialization Fixup

### PR DESCRIPTION
**Pull request by PKJosh**
Rather than commit straight to @mbayerPK 's branch, I want to retain a clear history of what changes I made and provide commentary on why they were made so that he can more easily digest the changes from his initial approach.

### CODE REVIEW

File: file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemTests.kt [hunk 1]
```diff
@@ -1191,6 +1016,48 @@ class AttributeCachingFileSystemTests {
         assertThat(Files.isHidden(cachingPath)).isEqualTo(expectedHidden)
     }
 
+    @Test
+    fun `attributes are not transferred when copying across filesystems`() {
```
[the remark this answers is not in the document]
Comment (PKJosh) [reply file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemTests.kt:1020]: Great point! Yeah, I think I was a little too lazy in proving the point, knowing the underlying the implementation and that I've got a solution that works. I'll rework these tests!

[the remark this answers is not in the document]
Comment (PKJosh) [reply file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemTests.kt]: Let me know what you think about the changes!

File: file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/LazyAttributeTest.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package com.pkware.filesystem.attributecaching
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.jupiter.api.Test
+
+class LazyAttributeTest {
```
[the remark this answers is not in the document]
Comment (PKJosh) [reply file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/LazyAttributeTest.kt:6]: Let me know what you think about the changes!

File: file-attribute-caching/src/main/kotlin/com/pkware/filesystem/attributecaching/LazyAttribute.kt [hunk 3]
```diff
@@ -4,63 +4,55 @@ package com.pkware.filesystem.attributecaching
...
-    private var lazyValue: T? = null
-
     /**
-     * Shows whether the [value] is initialized or not. `true` for initialized, default is `false`.
+     * Shows whether the [value] has been assigned. `true` for initialized/assigned, default is `false`.
      *
-     * If a value is initialized that means it is not `null`, without having to check/access it directly, as performing
-     * that action can trigger the [initializer] function to be called.
+     * If a value is assigned that means it has either had its initializer called via the invocation of its getter
+     * or its setter.
```
[the remark this answers is not in the document]
Comment (PKJosh) [reply file-attribute-caching/src/main/kotlin/com/pkware/filesystem/attributecaching/LazyAttribute.kt]: Let me know what you think about the changes!

**Review by TC-Iskandar — APPROVED** [review]

**Review by kirale18 — APPROVED** [review]

**Comment by PKJosh** [comment]
Covered by #13

**Pull request closed without merging**